### PR TITLE
add starlette-csrf 2.0.0

### PR DESCRIPTION
**Pull request by bollwyvl**
Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [ ] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [ ] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [ ] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by bollwyvl** [comment]
@conda-forge/help-python @conda-forge/staged-recipes ready for review :heart_eyes_cat:

**Comment by bollwyvl** [comment]
(sry, was trigger-happy: all CI has _actually_ :heavy_check_mark: now)

File: recipes/starlette-csrf/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  name: starlette-csrf
+  version: {{ version }}
+
+source:
+  url: https://pypi.io/packages/source/s/starlette-csrf/starlette_csrf-{{ version }}.tar.gz
+  sha256: 5f03b8c496840d612f1f7065aac7a209da084972f6e7844bcc5752c32ac54747
+
+build:
+  noarch: python
+  script: {{ PYTHON }} -m pip install . -vv --no-deps --no-build-isolation
```
Comment (ocefpaf) [inline recipes/starlette-csrf/meta.yaml:13]: You don't need the `--no-deps --no-build-isolation` but it doesn't hurt. (You do need them when using multiple outputs though but that is more a bug in our toolchain than a real requirement.)

File: recipes/starlette-csrf/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+test:
+  source_files:
+    - tests
+  imports:
+    - starlette_csrf
+  commands:
+    - pip check
+    - mypy -p starlette_csrf
+    - pytest -vv --cov=starlette_csrf --cov-branch --cov-report=term-missing:skip-covered --no-cov-on-fail --cov-fail-under=100
```
Comment (ocefpaf) [inline recipes/starlette-csrf/meta.yaml:35]: Coverage tests are probably not needed for a package. This is more a dev tool than a real testing of the package.

Comment (bollwyvl) [reply recipes/starlette-csrf/meta.yaml:35]: _probably_: but when i see an upstream that either reports _really high_ coverage, or weird import patterns, i try to put them in :blush:

**Pull request merged**